src/server/game/Entities/Unit/Unit.h

```diff
@@ -1193,12 +1195,17 @@ class TC_GAME_API Unit : public WorldObject
         CharmInfo* InitCharmInfo();
         void DeleteCharmInfo();
 
-        // all of these are for DIRECT CLIENT CONTROL only
-        void SetMovedUnit(Unit* target);
-        // returns the unit that this player IS CONTROLLING
-        Unit* GetUnitBeingMoved() const { return m_unitMovedByMe; }
-        // returns the player that this unit is BEING CONTROLLED BY
-        Player* GetPlayerMovingMe() const { return m_playerMovingMe; }
+        // base client control of this unit (possess effects, vehicles and similar). Not affected by temporary CC.
```

Comment:
I find this a bit concerning, meaning I at least dont know if this can cause any undesired effect.
Does the logic that uses these methods take into account the fact that the unit is actuallly still in client control, but temporally disabled due to a CC?
Is there a way to know if the unit is client controlled, even with any temporal CC aura?

Reply:
> Does the logic that uses these methods take into account the fact that the unit is actuallly still in client control, but temporally disabled due to a CC?

The PR aim is to not alter any behaviour in that regard. It is only to introduce functions that give the 'real time client control' status of a unit (IsMovedByClient() for example). These functions are needed to properly implement player movement.

> Is there a way to know if the unit is client controlled, even with any temporal CC aura?

Yes, by getting the "base client control" status of the unit; IsCharmerOrSelfPlayer()